core/chains/chain.go

```diff
@@ -129,6 +132,8 @@ func newValues(token Token) (int, error) {
 		currVal, err = Agoric()
 	case BNB:
 		currVal, err = Binance()
+	case EGLD:
```

Comment:
the original code could have been improved to work with interfaces & containers instead of this switch. Nothing to do